### PR TITLE
Dotenv ofn-install

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Provisions configs to a **Dotenv** file per environment

To test this in Staging we'll need to provision with this PR then deploy https://github.com/openfoodfoundation/openfoodnetwork/pull/7745 and see if the ENV vars are okay.

### CODE REVIEW

**Comment by sauloperez** [comment]
Why is dotenv needed?

**Comment by Matt-Yorkley** [comment]
It's just a proposal, but switching from Figaro to Dotenv in the main repo has various advantages: https://github.com/openfoodfoundation/openfoodnetwork/pull/7745#issue-662454878

I've started hitting cases where having all ENV vars for all environments in the same file at the same time with no flexibility is a real pain. I was looking at making it easier to run Sidekiq in dev, and looking at running `webpacker-dev-server` locally, and handling local Redis setups, etc. Having per-environment ENV vars that are easy to manage and apply local overrides for is really useful...

File: inventory/host_vars/_example.com/config.yml [hunk 1]
```diff
@@ -27,8 +27,8 @@ mail_domain: example.com
 #custom_hba_entries:
 #  - { type: hostssl, database: "{{ db }}", user: zapier, address: '54.86.9.50/32', auth_method: md5 }
 #
-#custom_application_env_vars: |
-#  OFN_FEATURE_CONNECT_AND_LEARN: "true"
+#custom_application_dotenv_vars: |
+#  OFN_FEATURE_CONNECT_AND_LEARN="true"
```
Comment (mkllnk) [inline inventory/host_vars/_example.com/config.yml:31]: I prefer to keep the old name. It's accurately describing the content without mentioning the implementation. We may go with another gem at some point. :shrug:

**Review by mkllnk — APPROVED** [review]

File: inventory/host_vars/_example.com/config.yml (hunk 1 above)
Comment (Matt-Yorkley) [reply inventory/host_vars/_example.com/config.yml:31]: Hmmm... I think it's still safest to use separate variables during the transition (we can delete the old one later). How about renaming `custom_application_dotenv_vars` to just `custom_env_vars`?

Comment (mkllnk) [reply inventory/host_vars/_example.com/config.yml]: Yes, I like that. We don't need a reference to `application.yml` any more. :+1:

**Review by mkllnk — APPROVED** [review]

**Review by andrewpbrett — APPROVED** [review]

**Comment by Matt-Yorkley** [comment]
Note; there's a build error here that's not related to these changes. Should be fixed by: https://github.com/openfoodfoundation/ofn-install/pull/755

**Pull request merged**